### PR TITLE
Fix: Currency form value reset on close

### PR DESCRIPTION
**Pull request by egor0798**

### CODE REVIEW

**Review by Asmadek — APPROVED** [review]

**Review by tuul-wq — CHANGES_REQUESTED** [review]

File: src/renderer/features/currency/CurrencyForm/ui/CurrencyForm.tsx [hunk 1]
```diff
@@ -77,7 +81,7 @@ export const CurrencyForm = ({ onSubmit }: Props) => {
         onChange={({ value }) => currency?.onChange(value.id)}
       />
 
-      <Button className="w-fit ml-auto" type="submit">
+      <Button className="w-fit ml-auto" type="submit" disabled={!currency?.isDirty && !fiatFlag?.isDirty}>
```
Comment (tuul-wq) [inline src/renderer/features/currency/CurrencyForm/ui/CurrencyForm.tsx:84]: UI should not know anything about business relations between fields
Computation for this flag should stay in the model
```javascript
const isFormValid = combine(...);
```

File: src/renderer/features/currency/CurrencyForm/model/currency-form.ts [hunk 2]
```diff
@@ -76,5 +80,6 @@ export const currencyFormModel = {
   $unpopularFiatCurrencies,
   events: {
     callbacksChanged: callbacksApi.callbacksChanged,
+    formInitiated,
```
Comment (tuul-wq) [inline src/renderer/features/currency/CurrencyForm/model/currency-form.ts:83]: Add test for this event 😊

File: src/renderer/features/currency/CurrencyForm/model/currency-form.ts [hunk 3]
```diff
@@ -32,17 +34,19 @@ const $unpopularFiatCurrencies = currencyModel.$currencyConfig.map((config) => {
...
+  fiatFlag: boolean | null;
+  currency: CurrencyItem | null;
+};
 
 sample({
-  clock: currencyModel.$activeCurrency,
-  filter: (currency): currency is CurrencyItem => currency !== null,
-  fn: (currency) => currency!.id,
-  target: $currencyForm.fields.currency.set,
+  clock: formInitiated,
```
Comment (tuul-wq) [inline src/renderer/features/currency/CurrencyForm/model/currency-form.ts:43]: What if `priceProviderModel.$fiatFlag` & `currencyModel.$activeCurrency` are still "loading"?
How can I set the form state when those stores will have finished?

**Review by tuul-wq — CHANGES_REQUESTED** [review]

File: src/renderer/features/currency/CurrencyForm/model/currency-form.ts [hunk 4]
```diff
@@ -32,17 +34,25 @@ const $unpopularFiatCurrencies = currencyModel.$currencyConfig.map((config) => {
   return config.filter((c) => c.category === 'fiat' && !c.popular);
 });
 
-sample({
-  clock: priceProviderModel.$fiatFlag,
-  filter: (fiatFlag): fiatFlag is boolean => fiatFlag !== null,
-  target: $currencyForm.fields.fiatFlag.set,
-});
+const $isFormValid = combine(
```
Comment (tuul-wq) [inline src/renderer/features/currency/CurrencyForm/model/currency-form.ts:37]: Super!

**Review by tuul-wq — APPROVED** [review]

**Pull request merged**